src/Workspaces/MSBuild/Core/MSBuild/BuildHostProcessManager.cs

```diff
@@ -382,12 +409,9 @@ public BuildHostProcess(Process process, string pipeName, ILoggerFactory? logger
 
             _process.ErrorDataReceived += Process_ErrorDataReceived;
 
-            var pipeClient = NamedPipeUtil.CreateClient(".", pipeName, PipeDirection.InOut, PipeOptions.Asynchronous);
+            // Inlined NamedPipeUtil.CreateClient because it does not support netstandard tfm due to using runtime specific functionality.
+            var pipeClient = new NamedPipeClientStream(".", GetPipeNameOrPath(pipeName), PipeDirection.InOut, PipeOptions.Asynchronous);
             pipeClient.Connect(TimeOutMsNewProcess);
-            if (!NamedPipeUtil.CheckPipeConnectionOwnership(pipeClient))
```

Comment:
How much would work it be to bring this back, to ensure we don't make a security mistake?

Reply:
On .NET and Mono this is a no-op. On Framework it would require pinvoking into Windows apis.

Reply:
Found the System.IO.Pipes.AccessControl package which makes the security side of this straightforward.